spyder/plugins/pythonpath/container.py

```diff
@@ -105,13 +105,14 @@ def update_active_project_path(self, path):
 
     def show_path_manager(self):
         """Show path manager dialog."""
-        # Set main attributes saved here
-        self.path_manager_dialog.update_paths(
-            self.path, self.not_active_path, get_system_pythonpath()
-        )
-
-        # Setup its contents again
-        self.path_manager_dialog.setup()
+        if not self.path_manager_dialog.isVisible():
```

Comment:
Why is this really necessary? And When could this happen?

I mean, since users can only have a single instance of the dialog, this should be necessary, right? So, to clarify why we need this to solve issue #20808, please add a comment about this line.

Reply:
When the icon is clicked, the action triggers `PythonpathContainer.show_path_manager` which _always_ calls `PathManager.update_paths` and `PathManager.setup`. If the widget is not already "open" then this is correct. But if it is already "open", then those methods will reset any changes the user may have made since opening the widget.

From what I could tell, the `PathManager` widget is instantiated on line 60 when `PythonpathContainer.setup` is called (I presume when the plugin is started at Spyder launch) and is never destroyed. The widget instance persists for the entire Spyder session, so the only thing to test for is whether it is visible or not.

So, #20808 revealed two problems in my opinion. 1) focus on the widget was not maintained after a directory was selected in the system file browser. 2) When clicking the icon, PYTHONPATH Manager _reset_ as well as restored focus; I think it should only restore focus if the widget is already "open".

Perhaps I'm confused and there is a better way to do this?

Reply:
> When the icon is clicked, the action triggers PythonpathContainer.show_path_manager which always calls PathManager.update_paths and PathManager.setup. If the widget is not already "open" then this is correct. But if it is already "open", then those methods will reset any changes the user may have made since opening the widget.

Ok, I see. So that's the problem.

> Perhaps I'm confused and there is a better way to do this?

I don't think so. Simply add a clarifying comment on top of this line to understand why the `if not self.path_manager_dialog.isVisible():` is necessary.